packages/polar/src/lib/deploy/contract.ts

```diff
@@ -0,0 +1,94 @@
+import fs from "fs-extra";
+import path from "path";
+import { CosmWasmClient, SigningCosmWasmClient } from "secretjs";
+
+import { PolarError } from "../../internal/core/errors";
+import { ERRORS } from "../../internal/core/errors-list";
+import {
+  ARTIFACTS_DIR,
+  CONTRACTS_DIR,
+  SCHEMA_DIR
+} from "../../internal/core/project-structure";
+import { Account, PolarRuntimeEnvironment } from "../../types";
+import { getClient, getSigningClient } from "../client";
+
+export class Contract {
+  readonly contract_path: string;
+  readonly schema_path: string;
+  readonly env: PolarRuntimeEnvironment;
+  readonly client: CosmWasmClient;
+  readonly signingClient: SigningCosmWasmClient;
+
+  codeId: number;
+  contractAddress: string;
+
+  constructor (contractName: string, account: Account, env: PolarRuntimeEnvironment) {
```

Comment:
I think we can have contractName in deploy function, then this class will become generalised for other contracts

Reply:
That way is fine too, but this way is better as user can easily keep track of contract object so that they can later query or execute through this object only. Also, user can differentiate contracts objects in this case, like contract1.getCount(), contract2.getOwner() etc. Simple for user.